deploy/role.yaml

```diff
@@ -22,6 +22,7 @@ rules:
   - watch
   - update
   - patch
+  - delete
```

Comment:
Noting that this should be transitional only and could be removed in a later version.

Reply:
I was going to file an issue about this, but then I realized that if someone was somehow running STF 1.5.2 and moved to STF 1.5.4 they would need this.

Additionally, the latest changes I landed in https://github.com/infrawatch/service-telemetry-operator/pull/475/commits/5d0210d1416681a102cba3cc097b18e74b22b927 will continually require the delete RBAC.

We _could_ refine this though and break out the `clusterroles` and `clusterrolebindings` though to have the `delete` resource only on the `clusterrolebindings`. I'm also curious if we could further lock this down to just resources we manage and not _all_ ClusterRoles and ClusterRoleBindings.